seatunnel-core/seatunnel-spark-starter/src/main/java/org/apache/seatunnel/core/starter/spark/config/SparkExecutionContext.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.core.starter.spark.config;
+
+import org.apache.seatunnel.apis.base.api.BaseSink;
+import org.apache.seatunnel.apis.base.api.BaseSource;
+import org.apache.seatunnel.apis.base.api.BaseTransform;
+import org.apache.seatunnel.core.starter.config.AbstractExecutionContext;
+import org.apache.seatunnel.core.starter.config.EngineType;
+import org.apache.seatunnel.core.starter.config.PluginType;
+import org.apache.seatunnel.plugin.discovery.PluginIdentifier;
+import org.apache.seatunnel.plugin.discovery.spark.SparkSinkPluginDiscovery;
+import org.apache.seatunnel.plugin.discovery.spark.SparkSourcePluginDiscovery;
+import org.apache.seatunnel.plugin.discovery.spark.SparkTransformPluginDiscovery;
+import org.apache.seatunnel.spark.SparkEnvironment;
+
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SparkExecutionContext extends AbstractExecutionContext<SparkEnvironment> {
+    private final SparkSourcePluginDiscovery sparkSourcePluginDiscovery;
+    private final SparkTransformPluginDiscovery sparkTransformPluginDiscovery;
+    private final SparkSinkPluginDiscovery sparkSinkPluginDiscovery;
+    private final List<URL> pluginJars;
+
+    public SparkExecutionContext(Config config, EngineType engine) {
+        super(config, engine);
+        this.sparkSourcePluginDiscovery = new SparkSourcePluginDiscovery();
+        this.sparkTransformPluginDiscovery = new SparkTransformPluginDiscovery();
+        this.sparkSinkPluginDiscovery = new SparkSinkPluginDiscovery();
+        List<URL> pluginJars = new ArrayList<>();
+        pluginJars.addAll(sparkSourcePluginDiscovery.getPluginJarPaths(getPluginIdentifiers(PluginType.SOURCE)));
+        pluginJars.addAll(sparkSinkPluginDiscovery.getPluginJarPaths(getPluginIdentifiers(PluginType.SINK)));
+        this.pluginJars = pluginJars;
+        this.getEnvironment().registerPlugin(pluginJars);
+    }
+
+    @Override
+    public List<BaseSource<SparkEnvironment>> getSources() {
```

Comment:
In our new spark starter module we will not use this class, BaseSource is used for old API.